src/main/java/org/hl7/davinci/priorauth/SubscriptionEndpoint.java

```diff
@@ -0,0 +1,147 @@
+            if (resource instanceof Subscription) {
+                Subscription subscription = (Subscription) resource;
+                if (processSubscription(subscription))
+                    formattedData = SUBSCRIPTION_ADDED_SUCCESS;
+                else {
+                    status = Status.BAD_REQUEST;
+                    OperationOutcome error = FhirUtils.buildOutcome(IssueSeverity.ERROR, IssueType.INVALID,
+                            PROCESS_FAILED);
+                    formattedData = requestType == RequestType.JSON ? App.getDB().json(error) : App.getDB().xml(error);
+                }
+            } else {
+                // Subscription is required...
+                status = Status.BAD_REQUEST;
+                OperationOutcome error = FhirUtils.buildOutcome(IssueSeverity.ERROR, IssueType.INVALID,
+                        REQUIRES_SUBSCRIPTION);
+                formattedData = requestType == RequestType.JSON ? App.getDB().json(error) : App.getDB().xml(error);
+            }
+        } catch (Exception e) {
+            // The subscription failed so spectacularly that we need to
+            // catch an exception and send back an error message...
+            status = Status.BAD_REQUEST;
+            OperationOutcome error = FhirUtils.buildOutcome(IssueSeverity.FATAL, IssueType.STRUCTURE, e.getMessage());
+            formattedData = requestType == RequestType.JSON ? App.getDB().json(error) : App.getDB().xml(error);
+        }
+        ResponseBuilder builder = requestType == RequestType.JSON
+                ? Response.status(status).type("application/fhir+json").entity(formattedData)
+                : Response.status(status).type("application/fhir+xml").entity(formattedData);
+        return builder.build();
+    }
+
+    private boolean processSubscription(Subscription subscription) {
+        // Get the criteria details parsed
+        String claimResponseId = "";
+        String patient = "";
+        String status = "";
+        String criteria = subscription.getCriteria();
+        String regex = "identifier=(.*)&patient.identifier=(.*)&status=(.*)";
+
+        Pattern pattern = Pattern.compile(regex);
+        Matcher matcher = pattern.matcher(criteria);
+
+        if (matcher.find() && matcher.groupCount() == 3) {
+            claimResponseId = matcher.group(1);
+            patient = matcher.group(2);
+            status = matcher.group(3);
+        } else {
+            logger.fine("Subscription.criteria: " + criteria);
+            logger.severe("Subcription.criteria is not in the form " + regex);
+            return false;
+        }
+
+        // Add to db
+        Map<String, Object> dataMap = new HashMap<String, Object>();
+        dataMap.put("claimResponseId", claimResponseId);
+        dataMap.put("patient", patient);
+        dataMap.put("status", status);
+        dataMap.put("resource", subscription);
+        return App.getDB().write(Database.SUBSCRIPTION, dataMap);
```

Comment:
This will add another subscription, are we notifying all?

Reply:
What do you mean by "another" subscription? This is where the subscription is added to the database. To create a subscription we post to this endpoint which will add the subscription details into the Subscription table. When a notification needs to be sent we search through this table for the notification details

Reply:
This does not notify all. The subscription resource is added to the table. When we need to send a notification we find this entry in the table. This PR only supports rest-hook so once we get the Subscription resource the notification is sent to the rest-endpoint provided in the Subscription resource

Reply:
What happens if the subscription endpoint is hit another time for the same claim? Will we only respond to the most recent request or will we send to both subscriptions?

Reply:
Oh I see what you mean. Yes it will only notify the first person

Reply:
Maybe we can address the multiple subscribers in the Spring based release.